### PR TITLE
Add yaml support

### PR DESCRIPTION
**Pull request by tim-heinsohn**
json is nice, yet sometimes it's useful to provide comments in config files ...

### CODE REVIEW

**Comment by igorw** [comment]
I do want to merge this, but some things need to be changed.
- I don't want to have a hard dependency on the YAML component. This should be changed to `suggest`.
- The version constraint on YAML should not be `dev-master`, but `2.1.*`.
- If YAML is selected but not available, an exception needs to be thrown.
- If the extension is not `yml`, `yaml` or `json`, an exception needs to be thrown.

**Comment by tim-heinsohn** [comment]
Agreed, please have another look.

**Comment by tim-heinsohn** [comment]
hey Igor, is there anything else you'd like to see changed?

File: src/Igorw/Silex/ConfigServiceProvider.php [hunk 1]
```diff
@@ -13,16 +13,49 @@
...
+     * format (json|yaml)
+     *
+     * @var string
+     */
+    private $format;
+
+    /**
+     * accepted file formats
+     *
+     * @var string
```
Comment (igorw) [inline src/Igorw/Silex/ConfigServiceProvider.php:37]: This is exactly the reason why I intentionally did not add doc blocks.

**Comment by igorw** [comment]
Merged, but I also changed/simplified some things, see 2fd96c9b19122c2d78da3949aa9b4bec60e90919.

**Comment by tim-heinsohn** [comment]
That's perfectly okay, thanks.

**Pull request merged**